grails-app/controllers/org/pih/warehouse/api/PutawayApiController.groovy

```diff
@@ -92,7 +92,7 @@ class PutawayApiController {
 
     private Putaway bindPutawayData(Putaway putaway, User currentUser, Location currentLocation, JSONObject jsonObject) {
         // Bind the putaway
-        bindData(putaway, jsonObject)
+        bindData(putaway, jsonObject, [exclude: ['putawayItems']])
```

Comment:
this is not the first time I came across this issue.
Before in grails 1 it didn't bind deeply nested items, which is why later in the code we manually iterate through `jsonObject.putawayItems` and bind them ourselves.
https://github.com/openboxes/openboxes/blob/e523fa3a9964532f1340b41080608c7f3402ae79/grails-app/controllers/org/pih/warehouse/api/PutawayApiController.groovy#L111-L123

Grails 3 apparently does handle properly binding nested objects, hence why there appears a problem of duplicated items: first set is from implicit binding and the other is from manual binding.
So we have to choose one or the other.

I chose to exclude the implicit binding and stick to the manual to avoid unnecessary regression.